### ISSUE TITLE
compile warning in org/osgi/test/common/context/CloseableBundleContextTest.java

### ISSUE THREAD

**Issue by bjhargrave** [osgi/osgi-test#131]
The CI build shows the following warning:

```
[WARNING] /home/runner/work/osgi-test/osgi-test/org.osgi.test.common/src/test/java/org/osgi/test/common/context/CloseableBundleContextTest.java:[159,60] unchecked conversion
  required: org.osgi.framework.ServiceReference<java.lang.Object>
  found:    org.osgi.framework.ServiceReference
```

**Comment by kriegfrj** [comment]
Interesting that the GUI does not warn about that one (it probably should).

**Comment by bjhargrave** [comment]
> Interesting that the GUI does not warn about that one (it probably should).

The joys of having 2 compilers: javac and ECJ. ECJ has many configurable settings. So it is possible that we could turn on some setting to make the warning. But I have observed over time that they are just different in what they accept. I have has generics which compiled fine in ECJ but were rejected by javac. So I had to rework the code to make both accept it.
